### DISCUSSION TITLE
DownloadTexture op JPG download bug

### DISCUSSION THREAD

**Discussion by mariomatasystems** [cables-gl/cables_docs#876] [! Welcome - readme first!]
Looking at the example for the downloadTexture op here: 
https://cables.gl/edit/15LaTs

When I download a JPG image I notice it doesn't open up in photoshop.

I also noticed that when I download an image with transparency it save the image as a .jpg but looking at it in the finder it still has transparency. maybe something to do with the meme type. Also think theres something that happens to the jpg download when the quality isn't set to 1 but I'm guessing here.

Opening it in photoshop I get this error:
'Could not open “screenshot (2).jpg” because an unknown or invalid JPEG marker type is found.'

**Comment by pandrr** [comment]
fixed

**Reply by mariomatasystems** [reply]
I noticed in your subpatch the number out of the ArrayContainsString op is going from 'found' rather than the 'index' value.

<img width="635" alt="Screenshot 2024-11-12 at 15 34 02" src="https://github.com/user-attachments/assets/b7adb500-62ce-4bcd-99de-6be4b9f6f95f">

**Comment by pandrr** [comment]
yep correct, thanks. fixed on dev.cables.gl
